src/styles/main.css

```diff
@@ -0,0 +1,85 @@
+body {
+  margin: 0;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: calc(700px + 6px);
+  position: relative;
+}
+
+.calendar__day {
+  box-sizing: border-box;
+  width: 100px;
+  height: 100px;
+  border: 1px solid #000;
+  background-color: #eee;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: relative;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day::before {
+  content: attr(data-day);
+  position: absolute;
+}
+
+.calendar__day:hover {
+  cursor: pointer;
+  background-color: #ffbfcb;
+  transform: translateY(-20px);
+  transition: transform 0.5s, background-color 0.5s;
+}
+
+.calendar--month-length-28 .calendar__day:nth-child(n+29) {
+  display: none;
+}
+
+.calendar--month-length-29 .calendar__day:nth-child(n+30) {
+  display: none;
+}
+
+.calendar--month-length-30 .calendar__day:nth-child(n+31) {
+  display: none;
+}
+
+.calendar--month-length-31 .calendar__day:nth-child(n+32) {
+  display: none;
+}
+
+.calendar--start-day-mon > .calendar__day:nth-child(1) {
+  margin-left: 0;
+}
+
+.calendar--start-day-tue > .calendar__day:nth-child(1) {
+  margin-left: 101px;
+}
+
+.calendar--start-day-wed > .calendar__day:nth-child(1) {
+  margin-left: 202px;
+}
+
+.calendar--start-day-thu > .calendar__day:nth-child(1) {
+  margin-left: 303px;
+}
+
+.calendar--start-day-fri > .calendar__day:nth-child(1) {
+  margin-left: 404px;
+}
+
+.calendar--start-day-sat > .calendar__day:nth-child(1) {
+  margin-left: 505px;
+}
+
+.calendar--start-day-sun > .calendar__day:nth-child(1) {
+  margin-left: 606px;
+}
```

Comment:
use scss loops instead repeat code.
do not hardcode px (pay attention on task description)
![Screenshot 2024-11-16 at 18 47 17](https://github.com/user-attachments/assets/2b7ecd8c-b534-4ed7-98b2-d21f6f225ade)